data_structures/stacks/postfix_evaluation.py

```diff
@@ -17,52 +24,205 @@
         Result =  59
 """
 
-import operator as op
+# Defining valid unary operator symbols
+UNARY_OP_SYMBOLS = ("-", "+")
+
+# Defining valid binary operator symbols
+BINARY_OP_SYMBOLS = ("-", "+", "*", "^", "/")
+
+
+def parse_token(token: str | float) -> float | str:
+    """
+    Converts the given data to appropriate number if it is indeed a number, else returns
+    the data as it is with a False flag. This function also serves as a check of whether
+    the input is a number or not.
+
+    Parameters
+    ----------
+    token : str or float
+        The data which needs to be converted to the appropriate number
+
+    Returns
+    -------
+    float or str
+        Returns a float if `token` is a number and returns `token` if it's an operator
+    """
+    if is_operator(token):
+        return token
+    try:
+        return float(token)
+    except ValueError:
+        msg = f"{token} is neither a number nor a valid operator"
+        raise ValueError(msg)
+
+
+def is_operator(token: str | float) -> bool:
+    """
+    Checks whether a given input is one of the valid operators or not.
+    Valid operators being '-', '+', '*', '^' and '/'.
+
+    Parameters
+    ----------
+    token : str
+        The value that needs to be checked for operator
+
+    Returns
+    -------
+    bool
+        True if data is an operator else False.
+    """
+    return token in BINARY_OP_SYMBOLS
+
+
+def evaluate(post_fix: list[str], verbose: bool = False) -> float:
+    """
+    Function that evaluates postfix expression using a stack.
+    >>> evaluate(["2", "1", "+", "3", "*"])
+    9.0
+    >>> evaluate(["4", "13", "5", "/", "+"])
```

Comment:
Please add doctests for:
* [ ] evaluate(["0"])
* [ ] evaluate(["-0"])
* [ ] evaluate(["1"])
* [ ] evaluate(["-1"])
* [ ] evaluate(["-1.1"])
* [ ] evaluate(["2", "1.9", "+", "3", "*"])
* [ ] evaluate(["2", "-1.9", "+", "3", "*"])
* [ ] evaluate(["2", "--1.9", "+", "3", "*"])